contrib/signet/addtxtoblock.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import sys
+import os
+import argparse
+
+# dirty hack but makes CBlock etc available
+sys.path.append('/'.join(os.getcwd().split('/')[:-2]) + '/test/functional')
+
+from test_framework.messages import (
+    CBlock,
+    CTransaction,
+    FromHex,
+    ToHex,
+)
+
+from test_framework.blocktools import add_witness_commitment
+
+def main():
+
+    # Parse arguments and pass through unrecognised args
+    parser = argparse.ArgumentParser(add_help=False,
+                                        usage='%(prog)s [addtxtoblock options] [bitcoin block file] [tx file] [fee]',
+                                        description=__doc__,
+                                        epilog='''Help text and arguments:''',
+                                        formatter_class=argparse.RawTextHelpFormatter)
+    # parser.add_argument('--filter', help='filter scripts to run by regular expression')
+    args, unknown_args = parser.parse_known_args()
```

Comment:
Idiomatic alternative which signals that `args` is intentionally unused: `_, unknown_args = parser.parse_known_args()` :)